apps/Ruffle/install

```diff
@@ -0,0 +1,129 @@
+    echo "Installing newer Rust toolchain"
+    wget -O /tmp/rustup-init.sh https://sh.rustup.rs
+    sudo chmod +x /tmp/rustup-init.sh
+    /tmp/rustup-init.sh -y
+    . "$HOME/.cargo/env"
+else
+    echo "Rust too old variable was not set or is at 0, continuing"
+fi
+
+# Download Ruffle source code and compile it
+git_clone https://github.com/ruffle-rs/ruffle || error 'Failed to clone Ruffle repository!'
+
+# Check if there is enough RAM available
+# Get free memory in MB
+free_mem=$(free -m | awk '/^Mem:/{print $7}')
+
+# Set the threshold (3GB = 3096MB)
+threshold=3096
+
+if (( free_mem < threshold )); then
+     warning "Warning: Free RAM is below 3GB ($free_mem MB). Compilation may fail unless swap/zram is available."
+     # Ask the user if they want to continue
+    read -p "Do you want to continue anyway even with not enough RAM available? Be warned that the compilation process could be killed by the OOM killer. (y/n): " choice
+
+    case "$choice" in
+        [Yy]*) echo "Continuing...";;
+        [Nn]*) echo "Exiting."; exit 1;;
+        *) echo "Invalid input. Exiting."; exit 1;;
+    esac
+fi
+
+status "Compiling Ruffle, this will take a bit"
+cd ruffle
+if cargo build --release --package=ruffle_desktop; then status_green "Compile success"; else error "Failed to compile Ruffle!"; fi
+
+# Copy compiled build
+status "Copying release to /opt/ruffle"
+cd target/release 
+mv ruffle_desktop ruffle || error "Failed to rename Ruffle binary!"
+# Only install back Ruffle binary if it's in update mode 
+if [ "$1" == update ];then
+  sudo cp ruffle /opt/ruffle || error "Failed to copy Ruffle binary to /opt/ruffle!"
+  exit 0
+fi
+sudo mkdir /opt/ruffle || error "Failed to make Ruffle folder in /opt!" 
+sudo cp ruffle /opt/ruffle || error "Failed to copy Ruffle binary to /opt/ruffle!" 
+sudo cp "$(dirname "$0")/icon-64.png" /opt/ruffle || error "Failed to copy Ruffle icon to /opt/ruffle!" 
+
+# Make command asociations
+status "Making terminal command..."
+echo '#!/bin/bash
+/opt/ruffle/ruffle "$@"' | sudo tee /usr/local/bin/ruffle >/dev/null
+sudo chmod +x /usr/local/bin/ruffle
+
+# Make menu launcher
+status "Making menu launcher..."
+echo "[Desktop Entry]
+Name=Ruffle
+Comment=Open source Flash Player emulator written in Rust
+Exec=ruffle %f
+Icon=/opt/ruffle/icon-64.png
+Terminal=false
+Type=Application
+Categories=Game;Utility;
+MimeType=application/x-shockwave-flash;" | sudo tee /usr/share/applications/ruffle.desktop >/dev/null
+
+# Make user associate .swf files to Ruffle
+description="Make MIME association for Ruffle?
+
+You will be able to launch .swf files without manually going into Ruffle and loading a Adobe Flash file within Ruffle's UI itself."
+userinput_func "$description" "Yes, associate Ruffle as the default application for .swf files." "No, don't associate Ruffle as the default application for .swf files."
+  if [ "$output" == "Yes, associate Ruffle as the default application for .swf files." ]; then
```

Comment:
This question will be asked during every update. For that reason, I try to pick sane defaults and I don't tend to ask about MIME attributions but just go ahead and do it. Why would a user not want this?

Reply:
Nvm, I saw line 77 now.